test/ModAttachr/modAttachr.ts

```diff
@@ -0,0 +1,36 @@
+import { ModAttachr } from "../../src/ModAttachr";
+
+import { mochaLoader } from "../main";
+
+import { EventNames } from "../../src/EventNames";
+
+import {
+    IMod
+} from "../../src/IModAttachr";
+
+mochaLoader.it("Creates a mod and fires the contained events", (): void => {
```

Comment:
Right now you're testing `onModEnable` and `onModDisable` in this test. That violates the concept of a "unit" test, which should only test one thing. Break this test up into two very similar tests. One should test `onModEnable` and one should test `onModDisable`. A third should also test some custom event.

When a test tests multiple things, and one of those things starts to fail in development, it's less clear that both are broken (since only the first error gets thrown). This makes it harder to TDD.